commit 83d435dd6e24b17955f91480f633bb39f0a7c2b4
Author: Xie XiuQi <xiexiuqi@huawei.com>
Date:   Fri Mar 29 09:43:47 2013 +0800

    Fix example error_injection_tool
    
    I got a "sched_setaffinity:: Invalid argument" error when using
    err_injection_tool to inject error on a system with over 32 cpus.
    
    Error information when injecting an error on a system with over 32 cpus:
    $ ./err_injection_tool -i
    /sys/devices/system/cpu/cpu0/err_inject//err_type_info
    Begine at Tue Mar 26 11:20:08 2013
    Configurations:
    On cpu32: loop=10, interval=5(s) err_type_info=4101,err_struct_info=95
    Error sched_setaffinity:: Invalid argument
    All done
    
    This because there is overflow when calculating the cpumask: the
    type of (1<<k) is int, while mask[j] is unsigned long. When k > 31,
    (1<<k) is truncated to ZERO, resulting in a unexpected cpumask.
    
    Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
    Signed-off-by: Tony Luck <tony.luck@intel.com>

diff --git a/Documentation/ia64/err_inject.txt b/Documentation/ia64/err_inject.txt
index 223e4f0582d0..9f651c181429 100644
--- a/Documentation/ia64/err_inject.txt
+++ b/Documentation/ia64/err_inject.txt
@@ -882,7 +882,7 @@ int err_inj()
 			cpu=parameters[i].cpu;
 			k = cpu%64;
 			j = cpu/64;
-			mask[j]=1<<k;
+			mask[j] = 1UL << k;
 
 			if (sched_setaffinity(0, MASK_SIZE*8, mask)==-1) {
 				perror("Error sched_setaffinity:");